hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTaskletBase.java

```diff
@@ -44,15 +47,18 @@
     final Processor processor;
     final OutboundEdgeStream[] outstreams;
     InboundEdgeStream currInstream;
+    private final JetInstance instance;
+    private final int processorIdx;
+    private ILogger logger;
 
-    private final Context context;
     private final ArrayDequeInbox inbox = new ArrayDequeInbox(progTracker);
     private final Queue<ArrayList<InboundEdgeStream>> instreamGroupQueue;
     private final String vertexName;
     private CircularListCursor<InboundEdgeStream> instreamCursor;
 
-    ProcessorTaskletBase(String vertexName, Processor.Context context, Processor processor,
-                         List<InboundEdgeStream> instreams, List<OutboundEdgeStream> outstreams) {
+    ProcessorTaskletBase(String vertexName, JetInstance instance, ILogger logger, Processor processor,
```

Comment:
It goes to the `Processor.Context`:

    void initProcessor(Outbox outbox, CompletableFuture<Void> jobFuture) {
        ProcCtx context = new ProcCtx(instance, logger, vertexName, processorIdx, jobFuture);
        processor.init(outbox, context);
    }